src/ShallowTraversal.js

```diff
@@ -127,9 +133,16 @@ export function buildPredicate(selector) {
       }
       break;
 
+    case 'object':
+      if (!Array.isArray(selector) && selector !== null && !isEmpty(selector)) {
+        return node => nodeMatchesObjectProps(node, selector);
+      }
+      throw new TypeError(
+        'Enzyme::Selector does not support an empty object, array, or null as a selector'
```

Comment:
I can't tell if you'd like me to just throw this in?

and if so is it best to just if..else if.. else if.. else it?
```
if (array) TypeError
else if (null) TypeError
else if (empty {}) TypeError
else return isMatch
```